pkg/helm/handlers/handlers.go

```diff
@@ -143,3 +149,70 @@ func (h *helmHandlers) HandleChartGet(user *auth.User, w http.ResponseWriter, r
 	res, _ := json.Marshal(resp)
 	w.Write(res)
 }
+
+
+func (h *helmHandlers) HandleHelmUpgradeRelease(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	var req HelmRequest
+
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to parse request: %v", err)})
+		return
+	}
+
+	conf := h.getActionConfigurations(h.ApiServerHost, req.Namespace, user.Token, &h.Transport)
+	resp, err := h.upgradeRelease(req.Namespace, req.Name, req.ChartUrl, req.Values, conf)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to upgrade helm release: %v", err)})
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	res, _ := json.Marshal(resp)
+	w.Write(res)
+}
+
+func (h *helmHandlers) HandleHelmUninstallRelease(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	params := r.URL.Query()
+	ns := params.Get("ns")
+	rel := params.Get("name")
+
+	conf := h.getActionConfigurations(h.ApiServerHost, ns, user.Token, &h.Transport)
+	resp, err := h.uninstallRelease(rel, conf)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to list helm releases: %v", err)})
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	res, _ := json.Marshal(resp)
+	w.Write(res)
+}
+
+func (h *helmHandlers) HandleHelmRollbackRelease(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	params := r.URL.Query()
+	ns := params.Get("ns")
+	rel := params.Get("name")
+
+	conf := h.getActionConfigurations(h.ApiServerHost, ns, user.Token, &h.Transport)
+	resp, err := h.rollbackRelease(rel, conf)
+	if err != nil {
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to list helm releases: %v", err)})
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	res, _ := json.Marshal(resp)
+	w.Write(res)
+}
+
+func (h *helmHandlers) HandleHelmPatchRelease(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	params := r.URL.Query()
+	revision := params.Get("revision")
```

Comment:
I would pass `revision` as the part of JSON request payload, and based on its existence or not decide if we are going to upgrade or rollback the given release.

Please add it here: https://github.com/akashshinde/console/blob/cdf6b189b71e488033ecaba7d90258d9f9453478/pkg/helm/handlers/request.go#L9-L8